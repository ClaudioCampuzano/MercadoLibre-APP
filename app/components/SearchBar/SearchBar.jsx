'use client';

import React, { useState } from 'react';
import styles from './SearchBar.module.scss';
import Link from 'next/link';
import { MercadoLibre, Search } from '@/public';
import Image from 'next/image';
import { useRouter } from 'next/navigation';

export const SearchBar = () => {
  const [product, setProduct] = useState('');
  const router = useRouter();

  const handleEnter = (e) => e.key === 'Enter' && handleSearchClick();
  const handleSearchClick = () => router.push(`/items?search=${product}`);

  return (
    <header className={styles.header}>
      <Link href="/">
        <Image src={MercadoLibre} alt="MercadoLibre" className={styles.imagenLogo} />
      </Link>
      <input
        placeholder="Busca productos, marcas y mÃ¡s..."
        className={styles.searchBox}
        onChange={(e) => setProduct(e.target.value)}
        value={product}
        onKeyDown={(e) => handleEnter(e)}
      />
      <button onClick={handleSearchClick}>
        <Image src={Search} alt="Search" className={styles.imagenSearch} />
      </button>
    </header>
  );
};
