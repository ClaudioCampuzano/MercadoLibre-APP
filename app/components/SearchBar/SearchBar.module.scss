@import '@/styles/constans.module.scss';

.header {
  height: 50px;
  width: 100%;
  background-color: $primaryColor;
  display: flex;
  align-items: center;
  justify-content: center;
}

.imagenLogo {
  width: 50px;
  height: 50px;
  padding-right: 10px;
}

.imagenSearch {
  width: 22px;
  height: 22px;
  cursor: pointer;
}

.searchBox {
  width: 70%;
  height: 50%;
}
