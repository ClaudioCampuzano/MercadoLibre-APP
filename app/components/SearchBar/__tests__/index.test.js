import React from 'react';

import { render } from '@testing-library/react';

import { SearchBar } from '../SearchBar';

jest.mock('next/navigation', () => ({
  useRouter() {
    return {
      route: '/',
      pathname: '',
      query: '',
      asPath: '',
    };
  },
}));

const componentRender = () => {
  return render(<SearchBar />);
};

describe('SearchBar', () => {
  it('renders', () => {
    const { container } = componentRender();

    expect(container).toMatchSnapshot();
  });
});
