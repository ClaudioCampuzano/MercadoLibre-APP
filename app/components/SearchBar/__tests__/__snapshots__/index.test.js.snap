// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders 1`] = `
<div>
  <header
    class="header"
  >
    <a
      href="/"
    >
      <img
        alt="MercadoLibre"
        class="imagenLogo"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </a>
    <input
      class="searchBox"
      placeholder="Busca productos, marcas y mÃ¡s..."
      value=""
    />
    <button>
      <img
        alt="Search"
        class="imagenSearch"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
    </button>
  </header>
</div>
`;
